source/common/http/http2/metadata_encoder.h

```diff
@@ -32,29 +34,26 @@ class MetadataEncoder : Logger::Loggable<Logger::Id::http2> {
   bool createPayload(const MetadataMapVector& metadata_map_vector);
 
   /**
-   * @return true if there is payload to be submitted.
+   * @return true if there is payload left to be packed.
    */
   bool hasNextFrame();
 
   /**
    * Creates the metadata frame payload for the next metadata frame.
    * @param buf is the pointer to the destination memory where the payload should be copied to. len
    * is the largest length the memory can hold.
-   * @return the size of frame payload.
+   * @return the size of frame payload, or -1 for failure.
    */
-  uint64_t packNextFramePayload(uint8_t* buf, const size_t len);
+  ssize_t packNextFramePayload(uint8_t* buf, const size_t len);
 
   /**
-   * Returns end_metadata value for the next metadata frame.
-   * @return end_metadata value.
+   * Returns a vector denoting the sequence of METADATA frames that this encoder expects to pack,
+   * and the flags to be set in each frame. This counts only frames that the encoder has not already
+   * packed; to get the full sequence of frames corresponding to the metadata map vector, call this
+   * before submitting any frames to nghttp2.
+   * @return The number of frames in each metadata payload, in sequence.
```

Comment:
"The number of frames in each metadata payload"
maybe 
"A vector indicating the number of frames in each payload?"
If it were the number of frames I'd expect
[2, 3]
instead of 
[0 0 [end] 0 0 0 [end]]
which I think is what we're getting.

Reply:
Oops, that line was for an old version of this function.

Your case there would actually be [0 [end] 0 0 [end]] - is that clear enough from this version?